### PR TITLE
Removed wordpress.sucuri.net API dependency

### PR DESCRIPTION
**Pull request by jacknm**
This PR removes the `https://wordpress.sucuri.net/api/` dependency, since this service has been discontinued until further notice. User who have their custom API to use in place of `https://wordpress.sucuri.net/api/` can still use the functionality by  adding the API endpoint as `SUCURISCAN_API_URL` on the `wp-config.php` file.

### CODE REVIEW

File: src/settings-apiservice.php [hunk 1]
```diff
@@ -30,41 +30,48 @@
...
 
-    if ($api_service === 'disabled') {
+    if ($api_service_option === 'disabled' || !$api_url_is_set) {
         $params['ApiStatus.StatusNum'] = '0';
         $params['ApiStatus.Status'] = __('Disabled', 'sucuri-scanner');
         $params['ApiStatus.SwitchText'] = __('Enable', 'sucuri-scanner');
         $params['ApiStatus.SwitchValue'] = 'enable';
         $params['ApiStatus.WarningVisibility'] = 'hidden';
         $params['ApiStatus.ErrorVisibility'] = 'visible';
+        $params['ApiStatus.ServiceURL'] = 'Service API URL not set. To enable the API service add your API service url as the SUCURISCAN_API_URL constant value to the main configuration file (wp-config.php).';
```
Comment (c-gvrentzos-godaddy) [inline src/settings-apiservice.php:65]:
```suggestion
        $params['ApiStatus.ServiceURL'] = 'Service API URL not set. To enable the API service, add your API service URL as the SUCURISCAN_API_URL constant value to the main configuration file (wp-config.php).';
```

File: .wp-env.json [hunk 2]
```diff
@@ -1,4 +1,7 @@
 {
     "core": null,
-    "plugins": ["."]
+    "plugins": ["."],
+    "config":{
+        "SUCURISCAN_API_URL": "https:://sucuri.net"
```
Comment (bfeddern-godaddy) [inline .wp-env.json:5]: Looks like you have two `:` here.  Not sure if that will mess with anything.

File: src/event.lib.php [hunk 3]
```diff
@@ -384,7 +384,7 @@ public static function sendLogToQueue($message = '')
      */
     public static function sendLogsFromQueue()
     {
-        if (SucuriScanOption::isDisabled(':api_service')) {
+        if (SucuriScanOption::isDisabled(':api_service') || !defined('SUCURISCAN_API_URL')) {
```
[the remark this answers is not in the document]
Comment (jacknm) [reply src/event.lib.php]: Thank you! Updated the check.

File: inc/tpl/settings-apiservice-status.html.tpl [hunk 4]
```diff
@@ -3,12 +3,10 @@
...
         <div class="sucuriscan-inline-alert-error sucuriscan-%%SUCURI.ApiStatus.ErrorVisibility%%">
-            <p>{{Disabling the API service communication will stop the event monitoring, consider to enable the <a href="%%SUCURI.URL.Settings%%#general">Log Exporter</a> to keep the monitoring working while the HTTP requests are ignored, otherwise an attacker may execute an action that will not be registered in the security logs and you will not have a way to investigate the attack in the future.}}</p>
+            <p>{{The API service is disabled consider enabling the <a href="%%SUCURI.URL.Settings%%#general">Log Exporter</a> to keep the monitoring working while the HTTP requests are ignored, otherwise an attacker may execute an action that will not be registered in the security logs and you will not have a way to investigate the attack in the future.}}</p>
```
[the remark this answers is not in the document]
Comment (jacknm) [reply inc/tpl/settings-apiservice-status.html.tpl]: I updated the wording here. It didn't make sense to me either.

File: .wp-env.json [hunk 5]
```diff
@@ -1,4 +1,7 @@
 {
     "core": null,
-    "plugins": ["."]
+    "plugins": ["."],
+    "config":{
+        "SUCURISCAN_API_URL": "https://sucuri.net"
```
Comment (c-gvrentzos-godaddy) [inline .wp-env.json:5]: Should `SUCURISCAN_API_URL` be set to `https://sucuri.net`?

Comment (jacknm) [reply .wp-env.json:5]: This is just a placeholder url for the e2e tests

File: sucuri.php [hunk 6]
```diff
@@ -8,7 +8,7 @@
  * Author: Sucuri Inc.
  * Text Domain: sucuri-scanner
  * Domain Path: /lang
- * Version: 1.8.36
+ * Version: 1.8.37
```
Comment (c-gvrentzos-godaddy) [inline sucuri.php:11]: Should we also update the `sucuri.php` and the `readme.txt` with the new version?

**Review by c-gvrentzos-godaddy — APPROVED** [review]
👍

**Pull request merged**